docs/adr/0015-separate-chain-service.md

```diff
@@ -0,0 +1,32 @@
+# 15. Chain Proxy as-a-Service
+
+Date: 2022-02-23
+
+## Status
+
+:hammer_and_wrench:
+
+## Context
+
+* To be able to scale our benchmarks and test Hydra Heads with more realistic configurations, we resurrected our `MockChain` component. While this is not strictly needed as we could also have set up a cardano private network with each Hydra node having its own Cardano node, it seemed much simpler and still provide us with the information we need as we are interested in the performance of the Head network, not the On-Chain part
+* This requires the ability to configure the `hydra-node` executable to use either a `MockChain` or `Direct` chain component to post and observe transactions, which means more command-line options and more code in the node
+* There is a clear separation of responsibilities, defined as a [component](./0007-with-pattern-component-interfaces.md) `ChainComponent`, between the core `Node` logic and the `Chain`. The interaction is mediated by "messages" representing the relevant transitions of the Hydra Head state machine, for posting transaction to the chain and observing them. This interface is completely agnostic of the actual chain semantics and, as hinted in the [Hydra paper]() could very well be implemented in a lot of different ways, and over a lot of different blockchains.
+* Different Hydra Head users could want to interact with the chain through different ways, whether it be [Blockfrost](https://blockfrost.io/), [Ogmios](https://ogmios.dev/), or a lightwallet, depending on their actual use cases, and this would require actually changing the Hydra node's code to implement a specific driver
+
+## Decision
+
+* The Hydra node _executable_ uses a `ChainComponent` that is implemented as a _generic client_ and configured using a single "port" (could be TCP or WS)
+* This client interacts with whatever _server_ it connects sending and receiving `PostChainTx` and `OnChainTx` messages in full-duplex mode
```

Comment:
So, what I get from this is that this _server_ wouldn't be generic but still quite tailored and aware of Hydra specifics; since it would have to yield `OnChainTx` and therefore, know how to construct / identify such transactions. 

Is it really what we want? I'd rather have that kind of logic lives inside the hydra-node. So perhaps, that chain server needs a slightly higher level interface to yield transactions or _some representation of transactions_ ? Or, are you already thinking long-term about possibly having this piece integrating with something else than Cardano.. thus not leaking Cardano specifics bits here and indeed, pack that transaction identification logic within the external chain-server :thinking:

Reply:
Yes. Post and OnChainTx are just a representation of the SM transition for the case of the cardano blockchain.

Reply:
Post- and OnChainTx are even parameterized by `tx` right now